src/collective/fingerpointing/logger.py

```diff
@@ -1,5 +1,55 @@
 # -*- coding: utf-8 -*-
-from collective.fingerpointing.config import PROJECTNAME
 import logging
+import time
+
+import zc.lockfile
+from collective.fingerpointing.config import PROJECTNAME
+from collective.fingerpointing.interfaces import IFingerPointingSettings
+from plone import api
+from plone.api.exc import InvalidParameterError
 
+HANDLER_REGISTERED = False
 logger = logging.getLogger(PROJECTNAME)
+
+
+def log_info(*args, **kwargs):
+    try:
+        record = IFingerPointingSettings.__identifier__ + '.audit_logfile'
+        logfile = api.portal.get_registry_record(record)
+    except InvalidParameterError:
+        logger.info(*args, **kwargs)
+        return
+
+    if logfile == '':
+        logger.info(*args, **kwargs)
+        return
+
+    if not globals()['HANDLER_REGISTERED']:
+        handler = logging.FileHandler(logfile)
+        formatter = logging.Formatter(
+            "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+        )
+        handler.setFormatter(formatter)
+
+        logger.addHandler(handler)
+
+    globals()['HANDLER_REGISTERED'] = True
+
+    # Try to lock the logfile then write to the logfile.
```

Comment:
if I understood correctly, you are trying to address here a multi-instance installation, right?